pkg/envoy/xds/ack.go

```diff
@@ -112,19 +118,50 @@ func (m *AckingResourceMutatorWrapper) Upsert(typeURL string, resourceName strin
 
 	version, _ := m.mutator.Upsert(typeURL, resourceName, resource, true)
 
-	comp := &pendingCompletion{
-		completion:              completion,
-		version:                 version,
-		typeURL:                 typeURL,
-		remainingNodesResources: make(map[string]map[string]struct{}, len(nodeIDs)),
-	}
+	comp, found := m.pendingCompletions[completion]
+	if found {
```

Comment:
Have you checked that this branch of the code is hit at all?

Reply:
New unit tests should cause this to be hit. I.e., original Completion completes successfully after a retry.

Reply:
Added a server_e2e_test that exercises this. Added a log line and verified that it is seen when the test is run.